core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixTextToSpeech.kt

```diff
@@ -0,0 +1,289 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.content.Context
+import android.media.AudioManager
+import android.media.AudioManager.OnAudioFocusChangeListener
+import android.speech.tts.TextToSpeech
+import android.speech.tts.TextToSpeech.Engine
+import android.speech.tts.TextToSpeech.LANG_MISSING_DATA
+import android.speech.tts.TextToSpeech.LANG_NOT_SUPPORTED
+import android.speech.tts.TextToSpeech.OnInitListener
+import android.speech.tts.TextToSpeech.QUEUE_ADD
+import android.speech.tts.TextToSpeech.SUCCESS
+import android.speech.tts.UtteranceProgressListener
+import android.util.Log
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import android.widget.Toast
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.toast
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.iSO3ToLocale
+import org.kiwix.kiwixmobile.core.utils.TAG_KIWIX
+import java.util.HashMap
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * Constructor.
+ *
+ * @param context the context to create TextToSpeech with
+ * @param onInitSucceedListener listener that receives event when initialization of TTS is done
+ * (and does not receive if it failed)
+ * @param onSpeakingListener listener that receives an event when speaking just started or
+ */
+class KiwixTextToSpeech internal constructor(
+  val context: Context,
+  onInitSucceedListener: OnInitSucceedListener,
+  val onSpeakingListener: OnSpeakingListener,
+  private val onAudioFocusChangeListener: OnAudioFocusChangeListener,
+  private val zimReaderContainer: ZimReaderContainer
+) {
+  private val focusLock: Any = Any()
+  private val am: AudioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
+  @JvmField var currentTTSTask: TTSTask? = null
+  private lateinit var tts: TextToSpeech
```

Comment:
inline `initTTS` here, this can become a val

Reply:
Should I use lazy?

Reply:
```  
private val tts: TextToSpeech by lazy { initTTS(onInitSucceedListener) }
private fun initTTS(onInitSucceedListener: OnInitSucceedListener): TextToSpeech =
    TextToSpeech(instance, OnInitListener { status: Int ->
      if (status == SUCCESS) {
        Log.d(TAG_KIWIX, "TextToSpeech was initialized successfully.")
        this.isInitialized = true
        onInitSucceedListener.onInitSucceed()
      } else {
        Log.e(TAG_KIWIX, "Initialization of TextToSpeech Failed!")
        context.toast(R.string.texttospeech_initialization_failed, Toast.LENGTH_SHORT)
      }
    })
```
Is this right?

Reply:
I've left a comment on the latest code so I'll resolve this